### PR TITLE
Update manual testing in DG

### LINKED ISSUE #177: Update instructions for manual testing in DG

### PR DESCRIPTION
**Pull request by tan-zhuoyao**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -778,25 +778,271 @@ testers are expected to do more *exploratory* testing.
...
+
+1. Adding a flashcard when in main window of application
+
+   1. Prerequisites: Application must be in main window and not in review or quiz mode.
+   
+   1. Test case (specifying compulsory inputs only): `add q/Does software projects often involve workflows? a/Yes` <br>
+       Expected: A new flashcard is added to the end of the list of flashcards. The compulsory inputs will be set to 
+       what was specified in the command, whereas category will be set to `General` by default when not stated in the
+       input. Result display will output the message: 
+       `New flashcard added:  Question: Does software projects often involve workflows?`
```
Comment (sgn00) [inline docs/DeveloperGuide.md]: Maybe can specify that all flashcards will be listed.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -778,25 +778,271 @@ testers are expected to do more *exploratory* testing.
...
+    1. Test case (view statistics of flashcard at invalid index): `stats 0` <br>
+        Expected: Flashcard view panel will not update and input text will turn red to signal an error.
+        Result display will output the invalid command error message.
+    
+    1. Other incorrect stats command to try: `stats x` (where x is larger than the list size or x is a negative integer) <br>
+        Expected: Similar to test case iii.
+        
 ### Saving data
 
 1. Dealing with missing/corrupted data files
```
Comment (sgn00) [inline docs/DeveloperGuide.md]: Will you be adding to this part?

**Review by sgn00 — CHANGES_REQUESTED** [review]
Refer comments

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (tan-zhuoyao) [reply docs/DeveloperGuide.md]: I have added additional details in

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (tan-zhuoyao) [reply docs/DeveloperGuide.md]: No. I have removed it

**Pull request merged**